assets/js/base/context/hooks/use-checkout-submit.js

```diff
@@ -54,16 +52,10 @@ export const useCheckoutSubmit = () => {
 	const waitingForProcessing =
 		isProcessing || isAfterProcessing || isBeforeProcessing;
 	const waitingForRedirect = isComplete && ! hasError;
-	const defaultLabel =
-		paymentMethod.placeOrderButtonLabel ||
-		__( 'Place Order', 'woo-gutenberg-products-block' );
-	const label = __experimentalApplyCheckoutFilter( {
-		filterName: 'placeOrderButtonLabel',
-		defaultValue: defaultLabel,
-	} );
+	const submitButtonText = paymentMethod.placeOrderButtonLabel || undefined;
```

Comment:
undefined is weird here, you chain with a truthy value not with a falsy value.
Is there a chance that `paymentMethod.placeOrderButtonLabel` would be something other than a full string or undefined?

Reply:
That was my bad. I've added `undefined` to having a return value in case `paymentMethod.placeOrderButtonLabel` has no value. I noticed that if `paymentMethod.placeOrderButtonLabel` has no value, it actually returns `undefined` already.